src/samplers/mongodbSampler.test.ts

```diff
@@ -0,0 +1,151 @@
+import { MongodbSampler, extractClientAttribute, matchMongoIsMaster } from './mongodbSampler';
+import { Context, SpanKind, Attributes, Link } from '@opentelemetry/api';
+import { SamplingDecision } from '@opentelemetry/sdk-trace-base';
+
+describe('MongodbSampler', () => {
+  let sampler: MongodbSampler;
+
+  beforeEach(() => {
+    sampler = new MongodbSampler();
+    delete process.env.LUMIGO_REDUCED_MONGO_INSTRUMENTATION;
+  });
+
+  it('should return RECORD_AND_SAMPLED when dbSystem and dbOperation are not provided', () => {
+    const result = sampler.shouldSample(
+      {} as Context,
+      'traceId',
+      'spanName',
+      SpanKind.CLIENT,
+      {},
+      []
+    );
+    expect(result.decision).toBe(SamplingDecision.RECORD_AND_SAMPLED);
+  });
+
+  it('should return NOT_RECORD when dbSystem is mongodb and dbOperation is isMaster and LUMIGO_REDUCED_MONGO_INSTRUMENTATION is true', () => {
+    process.env.LUMIGO_REDUCED_MONGO_INSTRUMENTATION = 'true';
```

Comment:
In general, I think it's better to split the tests where `process.env.LUMIGO_REDUCED_MONGO_INSTRUMENTATION` is `true` and `false`, and put those under two separate `describe` blocks, like this:
```ts
describe("when LUMIGO_REDUCED_MONGO_INSTRUMENTATION is 'true', () => {
    beforeEach(() => {
        process.env.LUMIGO_REDUCED_MONGO_INSTRUMENTATION = 'true';
    })

    // tests for `true`
})

describe("when LUMIGO_REDUCED_MONGO_INSTRUMENTATION is 'false', () => {
    beforeEach(() => {
        process.env.LUMIGO_REDUCED_MONGO_INSTRUMENTATION = 'false';
    })

    // tests for `false`
})
```